CRM/Price/Form/Set.php

```diff
@@ -197,36 +253,11 @@ public function buildQuickForm() {
       array('' => ts('- select -')) + $financialType, 'required'
```

Comment:
the code around financial type could be simplifed next round I think - the ACL is applied twice at the moment

Reply:
Yes, you can drop the entire `foreach` loop as it's doing exactly the same as `CRM_Financial_BAO_FinancialType::getIncomeFinancialType()` except for the `!in_array($key, $revenueFinancialType)` check but as it's unsetting array keys that makes no difference as the array keys will already have been dropped.  So I'd say just remove that foreach block now.